vic/drivers/cesm/cpl_mct/lnd_comp_mct.F90

```diff
@@ -0,0 +1,738 @@
+    USE seq_drydep_mod, ONLY: drydep_fields_token, lnd_drydep
+    USE shr_megan_mod , ONLY: shr_megan_fields_token, shr_megan_mechcomps_n
+
+    !--- ARGUMENTS:
+    IMPLICIT NONE
+
+    ! LOCAL VARIABLES:
+    TYPE(mct_aVect)   :: l2x      ! temporary, land to coupler
+    TYPE(mct_aVect)   :: x2l      ! temporary, coupler to land
+    TYPE(mct_aVect)   :: s2x      ! temporary, glacier to coupler
+    TYPE(mct_aVect)   :: x2s      ! temporary, coupler to glacier
+    INTEGER           :: num
+    CHARACTER(len= 2) :: cnum
+    CHARACTER(len=64) :: name
+    CHARACTER(len=32) :: subname = '(cpl_indices_set)'  ! subroutine name
+
+    !--- Determine attribute vector indices
+
+    !--- create temporary attribute vectors
+    CALL mct_aVect_init(x2l, rList=seq_flds_x2l_fields, lsize=1)
+    CALL mct_aVect_init(l2x, rList=seq_flds_l2x_fields, lsize=1)
+
+    !-------------------------------------------------------------
+    ! vic -> drv
+    !-------------------------------------------------------------
+    index_l2x_Flrl_rofliq   = mct_avect_indexra(l2x, 'Flrl_rofliq')
+    index_l2x_Flrl_rofice   = mct_avect_indexra(l2x, 'Flrl_rofice')
+
+    index_l2x_Sl_t          = mct_avect_indexra(l2x, 'Sl_t')
+    index_l2x_Sl_snowh      = mct_avect_indexra(l2x, 'Sl_snowh')
+    index_l2x_Sl_avsdr      = mct_avect_indexra(l2x, 'Sl_avsdr')
+    index_l2x_Sl_anidr      = mct_avect_indexra(l2x, 'Sl_anidr')
+    index_l2x_Sl_avsdf      = mct_avect_indexra(l2x, 'Sl_avsdf')
+    index_l2x_Sl_anidf      = mct_avect_indexra(l2x, 'Sl_anidf')
+    index_l2x_Sl_tref       = mct_avect_indexra(l2x, 'Sl_tref')
+    index_l2x_Sl_qref       = mct_avect_indexra(l2x, 'Sl_qref')
+    index_l2x_Sl_u10        = mct_avect_indexra(l2x, 'Sl_u10')
+    index_l2x_Sl_ram1       = mct_avect_indexra(l2x, 'Sl_ram1')
+    index_l2x_Sl_fv         = mct_avect_indexra(l2x, 'Sl_fv')
+    index_l2x_Sl_logz0      = mct_avect_indexra(l2x, 'Sl_logz0')
+    IF (lnd_drydep) THEN
+       index_l2x_Sl_ddvel = mct_avect_indexra(l2x, TRIM(drydep_fields_token))
+    ELSE
+       index_l2x_Sl_ddvel = 0
+    ENDIF
+
+    index_l2x_Fall_taux     = mct_avect_indexra(l2x, 'Fall_taux')
+    index_l2x_Fall_tauy     = mct_avect_indexra(l2x, 'Fall_tauy')
+    index_l2x_Fall_lat      = mct_avect_indexra(l2x, 'Fall_lat')
+    index_l2x_Fall_sen      = mct_avect_indexra(l2x, 'Fall_sen')
+    index_l2x_Fall_lwup     = mct_avect_indexra(l2x, 'Fall_lwup')
+    index_l2x_Fall_evap     = mct_avect_indexra(l2x, 'Fall_evap')
+    index_l2x_Fall_swnet    = mct_avect_indexra(l2x, 'Fall_swnet')
+    index_l2x_Fall_flxdst1  = mct_avect_indexra(l2x, 'Fall_flxdst1')
+    index_l2x_Fall_flxdst2  = mct_avect_indexra(l2x, 'Fall_flxdst2')
+    index_l2x_Fall_flxdst3  = mct_avect_indexra(l2x, 'Fall_flxdst3')
+    index_l2x_Fall_flxdst4  = mct_avect_indexra(l2x, 'Fall_flxdst4')
+
+    index_l2x_Fall_fco2_lnd = mct_avect_indexra(l2x, 'Fall_fco2_lnd', perrwith='quiet')
```

Comment:
Note added.  Also see reference [here](http://www.mcs.anl.gov/research/projects/mct/mct_APIs/node4.html) in section 2.1.11:

 > 2.1.11 indexRA_ - Index a Real Attribute
> 
> This function returns an INTEGER, corresponding to the location of a real attribute within the input AttrVect argument aV. For example, suppose aV has the following attributes 'latitude', 'longitude', and 'pressure'. The array of real values for the attribute 'longitude' is stored in aV%iAttr(indexRA_(aV,'longitude'),:). If indexRA_() is unable to match item to any of the real attributes in aV, the resulting value is zero which is equivalent to an error. The optional input CHARACTER arguments perrWith and dieWith control how such errors are handled.
> 
>    if neither perrWith nor dieWith are present, indexRA_() terminates execution with an internally generated error message;
>    if perrWith is present, but dieWith is not, an error message is written to stderr incorporating user-supplied traceback information stored in the argument perrWith;
>    if perrWith is present, but dieWith is not, and perrWith is equal to ``quiet'', no error message is written.
>    if dieWith is present, execution terminates with an error message written to stderr that incorporates user-supplied traceback information stored in the argument dieWith; and
>    if both perrWith and dieWith are present, execution terminates with an error message using dieWith, and the argument perrWith is ignored.
> 
> 
> INTERFACE:
> 
> 
>  integer function indexRA_(aV, item, perrWith, dieWith)